Sources/Hummingbird/Server/Request.swift

```diff
@@ -100,9 +119,29 @@ public struct HBRequest: Sendable, HBSendableExtensible {
     public func decode<Type: Decodable>(as type: Type.Type) throws -> Type {
         do {
             return try self.application.decoder.decode(type, from: self)
-        } catch {
+        }
+        catch DecodingError.dataCorrupted(_) {
+            let message = "The given data was not valid input."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch let DecodingError.keyNotFound(key, _) {
+            let path = key.pathKeyValue
+            let message = "Coding key `\(path)` not found."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch let DecodingError.valueNotFound(_, context) {
+            let path = context.codingPath.pathKeyValue
+            let message = "Value not found for `\(path)` key."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch let DecodingError.typeMismatch(type, context)  {
+            let path = context.codingPath.pathKeyValue
+            let message = "Type mismatch for `\(path)` key, expected `\(type)` type."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch {
             self.logger.debug("Decode Error: \(error)")
-            throw HBHTTPError(.badRequest)
+            throw HBHTTPError(.badRequest, message: error.localizedDescription)
```

Comment:
You should probably use `"\(error)"` here. You get a much better description than you do from `error.localizedDescription` usually.

Reply:
Yeah, in this case it makes sense. It's better to display more info.